src/memory-analyzer/memory_analyzer_parse_options.cpp

```diff
@@ -1,102 +1,188 @@
-  // config is later used to determine right size for types.
-  // If config is not set, you might see a bunch of types with
-  // size 0.
-  // It might be desierable to convert this into flags in the future.
-  config.ansi_c.mode = configt::ansi_ct::flavourt::GCC;
-  config.ansi_c.set_arch_spec_x86_64();
+  if(cmdline.isset("version"))
+  {
+    message.status() << CBMC_VERSION << '\n';
+    return CPROVER_EXIT_SUCCESS;
+  }
+
+  config.set(cmdline);
 
   if(cmdline.args.size() != 1)
   {
-    error() << "Please provide one binary with symbols to process" << eom;
-    return CPROVER_EXIT_USAGE_ERROR;
+    throw invalid_command_line_argument_exceptiont(
+      "no binary provided for analysis", "<binary>");
+  }
+
+  if(!cmdline.isset("symbols"))
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "need to provide symbols to analyse via --symbols", "--symbols");
+  }
+
+  const bool core_file = cmdline.isset("core-file");
+  const bool breakpoint = cmdline.isset("breakpoint");
+
+  if(!(core_file ^ breakpoint))
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "need to provide either option --core-file or option --breakpoint", "");
+  }
+
+  const bool output_file = cmdline.isset("output-file");
+  const bool symtab_snapshot = cmdline.isset("symtab-snapshot");
+
+  if(symtab_snapshot && output_file)
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "printing to a file is not supported for symbol table snapshot output",
+      "--symtab-snapshot");
   }
 
-  std::string binary = cmdline.args[0];
+  register_language(new_ansi_c_language);
 
-  gdb_apit gdb_api = gdb_apit(binary.c_str());
+  std::string binary = cmdline.args.front();
+
+  gdb_apit gdb_api(binary.c_str());
   gdb_api.create_gdb_process();
 
-  if(cmdline.isset("core-file"))
+  if(core_file)
   {
     std::string core_file = cmdline.get_value("core-file");
     gdb_api.run_gdb_from_core(core_file);
   }
-  else if(cmdline.isset("breakpoint"))
+  else if(breakpoint)
   {
     std::string breakpoint = cmdline.get_value("breakpoint");
     gdb_api.run_gdb_to_breakpoint(breakpoint);
   }
-  else
-  {
-    error() << "Either the 'core-file' or 'breakpoint; flag must be provided\n"
-            << "Cannot execute memory-analyzer. Going to shut it down...\n"
-            << messaget::eom;
 
-    gdb_api.terminate_gdb_process();
+  const std::string symbol_list(cmdline.get_value("symbols"));
+  std::vector<std::string> result;
+  split_string(symbol_list, ',', result, true, true);
 
-    help();
-    return CPROVER_EXIT_USAGE_ERROR;
-  }
+  auto opt = read_goto_binary(binary, ui_message_handler);
 
-  if(cmdline.isset("symbols"))
+  if(!opt.has_value())
   {
-    const std::string symbol_list(cmdline.get_value("symbols"));
-    std::vector<std::string> result;
-    split_string(symbol_list, ',', result, false, false);
+    throw deserialization_exceptiont(
+      "cannot read goto binary `" + binary + "`");
```

Comment:
:pick: :pick: :pick: The usual format for quoting we use is \` ', not \`\`

Reply:
Ok.